examples/utils/example-utils/src/migrator/sameContainerMigrator.ts

```diff
@@ -73,221 +69,197 @@ export class SameContainerMigrator
-						);
-					} catch {
-						// TODO: This implies that the contract is to throw if the data can't be transformed, which
-						// isn't great.  How should the dataTransformationCallback indicate failure?
-						this.emit("migrationNotSupported", acceptedVersion);
-						this._migrationP = undefined;
-						return;
-					}
-				} else {
-					// We can't get the data into a format that we can import, give up.
-					this.emit("migrationNotSupported", acceptedVersion);
-					this._migrationP = undefined;
-					return;
-				}
-				await migratedModel.importData(transformedData);
-
-				// Store the detached model for later use and retry scenarios
-				this._preparedDetachedModel = detachedModel;
-			};
-
-			const completeTheMigration = async () => {
-				assert(
-					this._preparedDetachedModel !== undefined,
-					"this._preparedDetachedModel should be defined",
-				);
-
-				// TODO: This is substantially different now
-			};
-
-			// Prepare the detached model if we haven't already.
-			if (this._preparedDetachedModel === undefined) {
-				await prepareTheMigration();
-			}
-
-			// Ensure another client has not already completed the migration.
-			// TODO: Real stuff
-			// if (this.migrationState !== "migrating") {
-			// 	return;
-			// }
+		const migrationSupported = await this.modelLoader.supportsVersion(acceptedVersion);
+		if (!migrationSupported) {
+			this.emit("migrationNotSupported", acceptedVersion);
+			return;
+		}
+		// TODO: this should be a reload of the same container basically
+		const migratedId = "foobar";
+		const migrated = await this.modelLoader.loadExisting(migratedId);
+		// Note: I'm choosing not to close the old migratable here, and instead allow the lifecycle management
+		// of the migratable to be the responsibility of whoever created the Migrator (and handed it its first
+		// migratable).  It could also be fine to close here, just need to have an explicit contract to clarify
+		// who is responsible for managing that.
+		this._currentModel = migrated;
+		this._currentModelId = migratedId;
+		this.emit("migrated", migrated, migratedId);
+
+		// Reset retry values
+		this._acceptedVersion = undefined;
+		this._pausedModel = undefined;
+		this._exportedData = undefined;
+		this._transformedData = undefined;
+		this._preparedDetachedModel = undefined;
+	};
 
-			await completeTheMigration();
-		};
+	private readonly monitorMigration = async () => {
+		// Ensure we are connected
+		if (!this.currentModel.connected()) {
+			await new Promise<void>((resolve) => this.currentModel.once("connected", resolve));
+		}
+		// Ensure the migration tool has reached the "readyForMigration" stage
+		if (this.currentModel.migrationTool.migrationState !== "readyForMigration") {
+			await new Promise<void>((resolve) =>
+				this.currentModel.migrationTool.once("readyForMigration", resolve),
+			);
+		}
 
-		this.emit("migrating");
+		console.log("Migration stage: startMigration");
+		if (this._acceptedVersion === undefined) {
```

Comment:
Since this is now a one-way function, all these conditional checks should always be true.  Fine to leave as checks for now but maybe these become asserts eventually (or just removed).